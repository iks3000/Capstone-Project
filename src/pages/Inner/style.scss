@import '../../assets/styles/_variables.scss';


.inner {
    transition: all 0.3s ease;

    &.show-hide-panel {
        margin-left: -250px;

        .show-hide-btn {
            left: 0;
        }
    }

    .show-hide-btn {
        position: fixed;
        width: 30px;
        height: 30px;
        display: flex;
        justify-content: center;
        align-items: center;
        box-sizing: border-box;
        background-color: $oxfordBlue;
        top: 30px;
        left: 250px;
        border-radius: 0 4px 4px 0;
        padding: 5px;
        border: none;
        cursor: pointer;
        transition: all 0.3s ease;
        color: $white;
        z-index: 10;


        &:hover {
            color: $green;
        }
    }

    .content {
        margin-left: 250px;
        background-color: $white;
        width: auto;
        height: 100%;
        display: block;
        padding: 30px;
    }

}

@media only screen and (max-width: 600px) {
    .inner  {
        &.show-hide-panel {
            margin-left: -55px;
        }

        .show-hide-btn {
            left: 55px;
        }

        .content {
            margin-left: 55px;
            padding: 20px 30px;
        }
    }
}