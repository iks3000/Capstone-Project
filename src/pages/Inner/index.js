import { useEffect, useReducer } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBars } from "@fortawesome/free-solid-svg-icons";
import Panel from "../../components/Panel";
import Box from "../../components/Box";
import TimeLine from "../../components/TimeLine";
import Expertise from "../../components/Expertise";
import Skills from "../../components/Skills";
import Portfolio from "../../components/Portfolio";
import Address from "../../components/Address";
import Feedback from "../../components/Feedback";
import ScrollToTop from "../../components/ScrollToTop";
import User1 from "../../assets/images/users/user1.png";
import User2 from "../../assets/images/users/user2.jpg";
import "./style.scss";
import { useDispatch, useSelector } from "react-redux";
import { fetchEducations } from "../../app/features/education/educationSlice";
import { fetchSkills } from "../../app/features/skills/skillsSlice";

const Content = () => {
  const dispatch = useDispatch();
  const educations = useSelector((state) => state?.educations.educations);
  const loading = useSelector((state) => state?.educations.loading);


  /* istanbul ignore next */
  useEffect(() => {
    dispatch(fetchEducations());
    dispatch(fetchSkills());
  }, [dispatch]);
  return (
    //passing real data to Expertise component instead of hardcoded
    <div className="content">
      <Box
        title="About me"
        content="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque."
      />
      <TimeLine data={educations} loading={loading} />
      <Expertise data={[
          {
            date: "2013-2014",
            info: {
              company: "Google",
              job: "Front-end developer / php programmer",
              description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringil",
            },
          },
          {
            date: "2012",
            info: {
              company: "Twitter",
              job: "Web developer",
              description: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim",
            },
          },
        ]}
      />
      <Skills />
      <Portfolio />
      <Address />
      <Feedback data={[
          {
            feedback: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus.",
            reporter: {
              photoUrl: User1,
              name: "John Doe",
              citeUrl: "https://www.citeexample.com",
            },
          },
          {
            feedback: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque.",
            reporter: {
              photoUrl: User2,
              name: "Smith Anderson",
              citeUrl: "https://www.citeexample.com",
            },
          },
        ]}
      />
    </div>
  );
};

const Inner = () => {
  const [isShow, toggle] = useReducer((isShow) => !isShow);

  return (
    <div className={`inner ${isShow && "show-hide-panel"}`}>
      <FontAwesomeIcon
        className="show-hide-btn"
        icon={faBars}
        onClick={toggle}
      />
      <Panel />
      <Content />
      <ScrollToTop />
    </div>
  );
};

export default Inner;

