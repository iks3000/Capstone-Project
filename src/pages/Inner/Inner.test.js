import React from "react";
import { mount, shallow } from "enzyme";
import Inner from "./index";
import { createStore } from "redux";
import { Provider } from "react-redux";
import educationReducer, {fetchEducations} from "../../app/features/education/educationSlice";
import { StaticRouter } from "react-router";

it("should render Inner component", () => {
  expect(shallow(<Inner />)).toMatchSnapshot();
});

describe("<Inner /> unit test", () => {
  const getWrapper = (mockStore = createStore(educationReducer, {})) =>
    mount(
      <Provider store={mockStore}>
        <StaticRouter location="someLocation">
          <Inner />
        </StaticRouter>
      </Provider>
    );
  it("should dispatch the correct action on button click", () => {
    const mockStore = createStore(educationReducer, {});
    mockStore.dispatch = jest.fn();

    const wrapper = getWrapper(mockStore);
    wrapper.find("button").simulate("click");
    expect(mockStore.dispatch).toHaveBeenCalledWith(fetchEducations());
  });
});
