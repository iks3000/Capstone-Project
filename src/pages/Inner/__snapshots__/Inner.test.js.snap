// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Inner component 1`] = `
<div
  className="inner undefined"
>
  <FontAwesomeIcon
    border={false}
    className="show-hide-btn"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f0c9",
          "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
        ],
        "iconName": "bars",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
  <Panel />
  <Content />
  <ScrollToTop />
</div>
`;
